addons/binding/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/service/MainTVServerService.java

```diff
@@ -176,7 +176,7 @@ public void onServiceSubscribed(String service, boolean succeeded) {
     public void onValueReceived(String variable, String value, String service) {
 
         String oldValue = stateMap.get(variable);
-        if (value.equals(oldValue)) {
+        if (oldValue.equals(value)) {
```

Comment:
oldValue can potentially be null in that case you will get an NPE, I'm not sure whether I came up with this idea myself or my IDE suggested it ;-)